src/analyze/analyze-security.c

```diff
@@ -490,7 +490,8 @@ static bool syscall_names_in_filter(Set *s, bool whitelist, const SyscallFilterS
                 if (syscall[0] == '@') {
                         const SyscallFilterSet *g;
                         assert_se(g = syscall_filter_set_find(syscall));
-                        b = syscall_names_in_filter(s, whitelist, g);
+                        /* false for param whitelist to always get uninverted results */
+                        b = !syscall_names_in_filter(s, false, g);
```

Comment:
The change looks tricky for me... Not tested, but I guess the following may fix the issue.
```
return syscall_names_in_filter(s, whitelist, g);
```

Reply:
Yeah, I confirm that my suggestion also works for, at least, the example you showed in the above.

Reply:
I wonder if the logic could be simplified by dropping the parameter `whitelist` in `syscall_names_in_filter()`. It would just report whether the syscalls are included or not. Then the logic in `assess_system_call_filter()` needs to be matched. Changing bools `b` and `bad` to `included` in the functions would also make the logic clearer. What do you think? I'm having some problems with my compilation environment right now, but I'll try to test this later.

Reply:
Yeah, sounds nice.

Reply:
I didn't find a way for this simplification. The way of analyzing is different for whitelist vs. blacklist modes. In whitelist mode all system calls must be included in filter for it to be effective, in blacklist mode we want to know if any of the system calls is included for it not to be effective.

I pushed a new version with a bit different logic and added some explanatory comments, PTAL. I think your version could return too early and miss further bad entries.